src/__tests__/RESTController-test.js

```diff
@@ -432,6 +437,30 @@ describe('RESTController', () => {
     CoreManager.set('SERVER_AUTH_TOKEN', null);
   });
 
+  it('sends requestId header for idempotency', async () => {
+    CoreManager.set('IDEMPOTENCY', true);
+    const requestIdHeader = (header) => 'X-Parse-Request-Id' === header[0];
+    const xhr = {
+      setRequestHeader: jest.fn(),
+      open: jest.fn(),
+      send: jest.fn()
+    };
+    RESTController._setXHR(function() { return xhr; });
+    RESTController.request('GET', 'classes/MyObject', {}, {});
```

Comment:
Parse Server does not enforce idempotency for GET requests. Idempotency is enforced only on POST, PUT routes as GET, DELETE are already idempotent. However, the test case itself looks good.